Assets/Scripts/Modules/Turn/ITurnModule.cs

```diff
@@ -1,8 +1,17 @@
 ﻿/*
- * Authors: Xijie Guo, Isaiah Mann
+ * Authors: Xijie Guo
  * Description: An interface to handle the turn logic
  */
 
-public interface ITurnModule {
+public interface ITurnModule { 
 
+	void SwitchToMonTurn();
+
+	bool CheckMonsterAttack(IUnit monster, IUnit player);
+
+	bool CheckForLastMon(IUnit monster, IUnit player);
```

Comment:
What is this checking? Whether it's the last monster's turn? If so, why does this function need the player?